<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="4" passed="2" failed="2" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-04-20T21:44:57 BDT" name="Suite" finished-at="2024-04-20T21:45:46 BDT" duration-ms="49298">
    <groups>
    </groups>
    <test started-at="2024-04-20T21:44:57 BDT" name="Login tests" finished-at="2024-04-20T21:45:46 BDT" duration-ms="49298">
      <class name="TestCases.Web.tests.ProductTests">
        <test-method is-config="true" signature="setup()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:44:57 BDT" name="setup" finished-at="2024-04-20T21:45:01 BDT" duration-ms="4438" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testToVerifyAddToCart()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:01 BDT" name="testToVerifyAddToCart" description="Test to verify add to cart" finished-at="2024-04-20T21:45:11 BDT" duration-ms="9256" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testToVerifyAddToCart -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:11 BDT" name="tearDown" finished-at="2024-04-20T21:45:11 BDT" duration-ms="198" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testToVerifyAddToCart status=SUCCESS method=ProductTests.testToVerifyAddToCart()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:11 BDT" name="setup" finished-at="2024-04-20T21:45:15 BDT" duration-ms="4103" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testToVerifyAddToCart2()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:15 BDT" name="testToVerifyAddToCart2" description="Test to verify add to cart2" finished-at="2024-04-20T21:45:21 BDT" duration-ms="6185" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Add cart button is not clicked expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Add cart button is not clicked expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at AbstarctComponents.Web.FrameworkAssertionLibrary.stepAssertionShouldBeTrue(FrameworkAssertionLibrary.java:23)
at TestCases.Web.steps.TestSteps.clickingAddToCart2(TestSteps.java:39)
at TestCases.Web.tests.ProductTests.testToVerifyAddToCart2(ProductTests.java:17)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testToVerifyAddToCart2 -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:21 BDT" name="tearDown" finished-at="2024-04-20T21:45:21 BDT" duration-ms="164" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testToVerifyAddToCart2 status=FAILURE method=ProductTests.testToVerifyAddToCart2()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:21 BDT" name="setup" finished-at="2024-04-20T21:45:25 BDT" duration-ms="3279" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testToVerifyAddToCart3()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:25 BDT" name="testToVerifyAddToCart3" description="Test to verify add to cart" finished-at="2024-04-20T21:45:34 BDT" duration-ms="9235" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testToVerifyAddToCart3 -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:34 BDT" name="tearDown" finished-at="2024-04-20T21:45:34 BDT" duration-ms="189" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testToVerifyAddToCart3 status=SUCCESS method=ProductTests.testToVerifyAddToCart3()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:34 BDT" name="setup" finished-at="2024-04-20T21:45:40 BDT" duration-ms="5432" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="testToVerifyAddToCart4()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:40 BDT" name="testToVerifyAddToCart4" description="Test to verify add to cart2" finished-at="2024-04-20T21:45:46 BDT" duration-ms="6190" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Add cart button is not clicked expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Add cart button is not clicked expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at AbstarctComponents.Web.FrameworkAssertionLibrary.stepAssertionShouldBeTrue(FrameworkAssertionLibrary.java:23)
at TestCases.Web.steps.TestSteps.clickingAddToCart2(TestSteps.java:39)
at TestCases.Web.tests.ProductTests.testToVerifyAddToCart4(ProductTests.java:30)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testToVerifyAddToCart4 -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c]" started-at="2024-04-20T21:45:46 BDT" name="tearDown" finished-at="2024-04-20T21:45:46 BDT" duration-ms="174" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testToVerifyAddToCart4 status=FAILURE method=ProductTests.testToVerifyAddToCart4()[pri:0, instance:TestCases.Web.tests.ProductTests@33a2499c] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- TestCases.Web.tests.ProductTests -->
    </test> <!-- Login tests -->
  </suite> <!-- Suite -->
</testng-results>
